emergence_lib/src/organisms/organism_details.rs

```diff
@@ -0,0 +1,160 @@
+//! Detailed info about a given organism.
+
+use bevy::prelude::*;
+use bevy_ecs_tilemap::tiles::TilePos;
+
+use std::fmt::Display;
+
+use crate::{
+    cursor::CursorTilePos,
+    items::{Inventory, Recipe},
+};
+
+use super::{
+    structures::{
+        crafting::{
+            ActiveRecipe, CraftTimer, CraftingState, CurCraftState, InputInventory, OutputInventory,
+        },
+        fungi::Fungi,
+        plants::Plant,
+    },
+    units::Ant,
+};
+
+/// The type of the organism, e.g. plant or fungus.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum OrganismType {
+    /// A plant.
+    Plant,
+
+    /// A fungus.
+    Fungus,
+
+    /// An ant.
+    Ant,
+}
+
+impl Display for OrganismType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "{}",
+            match self {
+                OrganismType::Plant => "Plant",
+                OrganismType::Fungus => "Fungus",
+                OrganismType::Ant => "Ant",
+            }
+        )
+    }
+}
+
+/// The details about crafting processes.
+#[derive(Debug, Clone)]
+pub struct CraftingDetails {
+    /// The inventory for the input items.
+    pub input_inventory: Inventory,
+
+    /// The inventory for the output items.
+    pub output_inventory: Inventory,
+
+    /// The recipe that's currently being crafted.
+    pub active_recipe: Recipe,
+
+    /// The state of the ongoing crafting process.
+    pub state: CraftingState,
+
+    /// The time remaining to finish crafting.
+    pub timer: Timer,
+}
+
+/// Detailed info about a given entity.
+#[derive(Debug, Clone)]
+pub struct OrganismDetails {
+    /// The entity ID of the organism that this info is about.
+    pub entity: Entity,
+
+    /// The type of the organism, e.g. plant or fungus.
+    pub organism_type: OrganismType,
+
+    /// If this organism is crafting something, the details about that.
+    pub crafting_details: Option<CraftingDetails>,
+}
+
+/// Detailed info about the organism that is being hovered.
+#[derive(Debug, Resource, Default)]
+pub struct HoverDetails(pub Option<OrganismDetails>);
+
+/// Display detailed info on hover.
+pub struct DetailsPlugin;
+
+impl Plugin for DetailsPlugin {
+    fn build(&self, app: &mut App) {
+        info!("Building DetailsPlugin...");
+
+        app.init_resource::<HoverDetails>()
+            // TODO: This should be done after the cursor system
+            .add_system(hover_details);
+    }
+}
+
+/// Get details about the hovered entity.
+fn hover_details(
+    cursor_pos: Res<CursorTilePos>,
+    mut hover_details: ResMut<HoverDetails>,
+    query: Query<(
```

Comment:
Would it make sense to use `bevy-trait-query` here? https://crates.io/crates/bevy-trait-query

A `HoverDetails` trait for Plant/Fungi/Ant etc. could be implemented, and then queried for?

Reply:
Good question, might be possible!

But yea I think I'd prefer to push that to another PR :)